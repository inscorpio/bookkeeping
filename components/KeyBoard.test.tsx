import { beforeEach, describe, expect, it } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'
import Keyboard from './KeyBoard'

beforeEach(() => {
  render(<Keyboard />)
})

describe('keyboard', () => {
  const testId = 'value'

  it('should be render', async () => {
    const { container } = render(<Keyboard />)
    expect(container).toMatchSnapshot()
  })

  describe('click number', () => {
    it('should concat new number', async () => {
      await fireEvent.click(screen.getByText('1'))
      await fireEvent.click(screen.getByText('2'))
      expect(screen.getByTestId(testId).innerHTML).toBe('12')
    })
  })

  describe('click dot', () => {
    it('should maintain one dot when click one more time dot', async () => {
      await fireEvent.click(screen.getByText('1'))
      await fireEvent.click(screen.getByText('.'))
      await fireEvent.click(screen.getByText('.'))
      await fireEvent.click(screen.getByText('.'))
      expect(screen.getByTestId(testId).innerHTML).toBe('1.')
    })
  })

  describe('click icon', () => {
    it('should clear value when click clear icon', async () => {
      await fireEvent.click(screen.getByText('1'))
      await fireEvent.click(screen.getByText('2'))

      expect(screen.getByTestId(testId).innerHTML).toBe('12')

      await fireEvent.click(screen.getByTestId('icon-clear').parentElement!)
      expect(screen.getByTestId(testId).innerHTML).toBe('0')
    })

    it('should remove one char when click backspace icon', async () => {
      await fireEvent.click(screen.getByText('1'))
      await fireEvent.click(screen.getByText('2'))

      expect(screen.getByTestId(testId).innerHTML).toBe('12')

      await fireEvent.click(screen.getByTestId('icon-backspace').parentElement!)
      expect(screen.getByTestId(testId).innerHTML).toBe('1')
    })
  })
})
