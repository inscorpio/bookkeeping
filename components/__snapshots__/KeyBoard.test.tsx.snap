// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`keyboard > should be render 1`] = `
<div>
  <div
    class="space-y-2"
  >
    <ul
      class="flex items-center gap-2 p-2 w-full text-xs bg-white"
    >
      <li
        class="flex items-center gap-1"
      >
        <svg
          fill="currentColor"
          height="1.2em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1.2em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
        <span
          class="mt-[2px]"
        >
          今天
        </span>
      </li>
      <li
        class="ml-auto"
      >
        金额：
        <span
          class="text-base"
          data-testid="value"
        >
          0
        </span>
      </li>
    </ul>
    <ul
      class="grid grid-cols-4 gap-1"
    >
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        1
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        2
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        3
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        <svg
          data-testid="icon-backspace"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 4H8l-7 8 7 8h13a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2z"
          />
          <line
            x1="18"
            x2="12"
            y1="9"
            y2="15"
          />
          <line
            x1="12"
            x2="18"
            y1="9"
            y2="15"
          />
        </svg>
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        4
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        5
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        6
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white row-span-3"
      >
        保存
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        7
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        8
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        9
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        <svg
          data-testid="icon-clear"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          t="1569683368540"
          version="1.1"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M899.1 869.6l-53-305.6H864c14.4 0 26-11.6 26-26V346c0-14.4-11.6-26-26-26H618V138c0-14.4-11.6-26-26-26H432c-14.4 0-26 11.6-26 26v182H160c-14.4 0-26 11.6-26 26v192c0 14.4 11.6 26 26 26h17.9l-53 305.6c-0.3 1.5-0.4 3-0.4 4.4 0 14.4 11.6 26 26 26h723c1.5 0 3-0.1 4.4-0.4 14.2-2.4 23.7-15.9 21.2-30zM204 390h272V182h72v208h272v104H204V390z m468 440V674c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v156H416V674c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v156H202.8l45.1-260H776l45.1 260H672z"
          />
        </svg>
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        0
      </li>
      <li
        class="flex justify-center items-center py-2 rounded-sm bg-white text-base"
      >
        .
      </li>
    </ul>
  </div>
</div>
`;
